Issue/2652 unobscure password post options

diegoreymendez opened the pull request:
Fixes #2652.

Also fixes an issue that was causing the password to be reset when the device was rotated.  This can be reproduce in develop adding a password to a post (as mentioned in the steps below) and then rotating the device.

**To test:**

- Create a new post in the app.
- Go to the post options (top-right corner "..." then select "Options").
- Under the visibility option pick: "Password Protected".
- Enter a password.
- Show and hide it using the added button.

frosty commented on WordPress/Classes/System/Constants.m:
```diff
@@ -24,6 +24,11 @@
 
+/// Image file names
+///
+NSString *const WPIconShowPassword                                  = @"icon-secure-text-visible";
```
Can we use the `visible` and `not-visible` Gridicons here instead of images? Or are they not the same as the ones we're using?

diegoreymendez replied:
I can check them out and remove the pre-existing local image in the process.

frosty commented on WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m:
```diff
@@ -584,25 +613,44 @@ - (UITableViewCell *)configureMetaPostMetaCellForIndexPath:(NSIndexPath *)indexP
+- (void)configureVisibilityButtonForPasswordCell:(WPTextFieldTableViewCell *)textCell
+{
+    self.passwordVisibilityButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    self.passwordVisibilityButton.frame = CGRectMake(0, 0, 24, 20);
```
Can we pass the image size here instead of using magic numbers? Or will the button size to fit its image anyway?

frosty commented:
Hey @diegoreymendez! Changes look good, except two small comments and: the button state is the opposite of what we use on the login screen. So on login, if the button shows the open eye then the password is visible. If it shows the closed eye, the password is hidden. I think that way of doing it makes the most sense.

diegoreymendez commented:
@frosty - Good feedback.

These are my changes to address your comments:

- Changed the icons to load from Gridicons.
- Removed the constants I had added.
- Changed `WPWalkthroughTextField.m` to also use the Gridicons icons.  You can test this by trying to add a self-hosted site to the app.
- Deleted the local images we had for those two icons.

So the PR is ready for round 2.  Let me know.

frosty commented on WordPress/Classes/ViewRelated/NUX/WPWalkthroughTextField.m:
```diff
@@ -57,8 +59,8 @@ - (void)commonInit
+    self.secureTextEntryImageHidden = [Gridicon iconOfType:GridiconTypeNotVisible];
 
     self.secureTextEntryToggle = [UIButton buttonWithType:UIButtonTypeCustom];
     self.secureTextEntryToggle.frame = CGRectMake(0, 0, 40, 30);
```
Can you also remove the manual frame setting here?

diegoreymendez replied:
Considering this is outside the scope of the PR, can you specify the change you'd like here?  I'd like to avoid a trial and error.

If I adjust the button to the default Gridicon size, the new code is going to have a different size.

PS: I don't think it'd be wise to continue extending this PR, but I don't mind changing it if the request is specific.

frosty commented:
Changes look good, @diegoreymendez! One final tweak (in addition to the code comment I just added): 

t looks like the button needs to have a different `tintColor` set – the original image must've been colored instead of tinted. I think `greyLighten10` should do it (needs to be set for both sign in and the post settings).

diegoreymendez commented:
Adjusted the tint color.  Ready for another look.

frosty commented:
:shipit:!

diegoreymendez commented:
Thanks James!
